### ISSUE TITLE
exercise grains overflows unsigned long long

### ISSUE THREAD

**Issue by JAC42** [exercism/exercism.io#2176]
In C++, the test for total number of grains merely tests to see whether you get the maximum value for unsigned long long, not for the correct number of grains.  I believe 2^64 is max number allowed in unsigned long long, and the total grains on the board is greater than that given 64 squares.

**Comment by kytrinyx** [comment]
I think you're right. I've moved this issue into the C++ track repository: https://github.com/exercism/xcpp/issues/35

**Comment by LegalizeAdulthood** [comment]
If the C++ test is wrong, then the [JavaScript test](https://github.com/exercism/xjavascript/blob/master/grains/grains_test.spec.js) is also wrong.

Why is 18446744073709551615 the wrong answer?

**Comment by JAC42** [comment]
It is the wrong answer because it is merely the largest value that unsigned long long will hold, not that actual number of grains on the whole board.  Go here: http://www.cplusplus.com/reference/climits/ and look at the max for unsigned long long.  It's 18446744073709551615.

**Comment by LegalizeAdulthood** [comment]
I understand that value is 2^64-1.  My question is why you think that is the wrong answer for the test.  The JavaScript test for this problem is using the same value, as does [Ruby](https://github.com/exercism/xruby/blob/master/grains/grains_test.rb), [perl5](https://github.com/exercism/xperl5/blob/master/grains/cases.json), [scala](https://github.com/exercism/xscala/blob/master/grains/src/test/scala/grains_test.scala), [Clojure](https://github.com/exercism/xclojure/blob/master/grains/grains_test.clj), [Elixir](https://github.com/exercism/xelixir/blob/master/grains/grains_test.exs), [F#](https://github.com/exercism/xfsharp/blob/master/grains/GrainsTest.fs), [Go](https://github.com/exercism/xgo/blob/master/grains/grains_test.go), [Haskell](https://github.com/exercism/xhaskell/blob/master/grains/grains_test.hs), etc.

Basically every single language track that implements this problem uses this exact same value for this test case.

**Comment by JAC42** [comment]
Maybe I'm wrong.  Mathematics tells us (http://en.wikipedia.org/wiki/Geometric_progression#Geometric_series) that the answer is 2^64 - 1.  I guess it is just a coincidence that it is the maximum value for unsigned long long.  Sorry for any mixup.
